.clang-format

```diff
@@ -0,0 +1,219 @@
+MacroBlockEnd: '^END_COM_MAP$|^END_CONNECTION_POINT_MAP$|^END_HELPER_NODEMAP$|^END_MODULE$|^END_MSG_MAP$|^END_OBJECT_MAP$|^END_TEST_CLASS$|^END_TEST_METHOD$|^END_TEST_METHOD_PROPERTIES$'
+MaxEmptyLinesToKeep: 1
+NamespaceIndentation: Inner
+ObjCBlockIndentWidth: 2
+ObjCSpaceAfterProperty: false
+ObjCSpaceBeforeProtocolList: true
+PenaltyBreakBeforeFirstCallParameter: 19
+PenaltyBreakComment: 300
+PenaltyBreakFirstLessLess: 120
+PenaltyBreakString: 2000
+PenaltyExcessCharacter: 2
+PenaltyReturnTypeOnItsOwnLine: 1000
+PointerAlignment: Left
+SortIncludes: false
+SpaceAfterCStyleCast: false
+SpaceBeforeAssignmentOperators: true
+SpaceBeforeParens: ControlStatements
+SpaceInEmptyParentheses: false
+SpacesBeforeTrailingComments: 1
+SpacesInAngles: Never
+SpacesInContainerLiterals: true
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+Standard: Latest
+TabWidth: 4
+UseTab: Never
+
+AttributeMacros: [
+  CALLBACK,
+]
+
+StatementMacros: [
+  _Acquires_exclusive_lock_,
+  _Acquires_lock_,
+  _Acquires_nonreentrant_lock_,
+  _Acquires_shared_lock_,
+  _Analysis_assume_smart_lock_acquired_,
+  _Analysis_assume_smart_lock_released_,
+  _Create_lock_level_,
+  _Detaches_lock_,
+  _Function_class_,
+  _Global_cancel_spin_lock_,
+  _Global_critical_region_,
+  _Global_interlock_,
+  _Global_priority_region_,
+  _Has_lock_kind_,
+  _Has_lock_level_,
+  _IRQL_always_function_max_,
+  _IRQL_always_function_min_,
+  _IRQL_raises_,
+  _IRQL_requires_,
+  _IRQL_requires_max_,
+  _IRQL_requires_min_,
+  _IRQL_requires_same_,
+  _IRQL_restores_,
+  _IRQL_restores_global_,
+  _IRQL_saves_,
+  _IRQL_saves_global_,
+  _Lock_level_order_,
+  _Moves_lock_,
+  _Must_inspect_result_,
+  _No_competing_thread_,
+  _Post_same_lock_,
+  _Post_writable_byte_size_,
+  _Pre_satisfies_,
+  _Releases_exclusive_lock_,
+  _Releases_lock_,
+  _Releases_nonreentrant_lock_,
+  _Releases_shared_lock_,
+  _Replaces_lock_,
+  _Requires_exclusive_lock_held_,
+  _Requires_lock_held_,
+  _Requires_lock_not_held_,
+  _Requires_no_locks_held_,
+  _Requires_shared_lock_held_,
+  _Ret_maybenull_,
+  _Ret_range_,
+  _Struct_size_bytes_,
+  _Success_,
+  _Swaps_locks_,
+  _Use_decl_annotations_,
+  _When_,
+
+  DECLARE_ORDINAL_MAP,
+  DECLARE_PROCNAME_MAP,
+  DEFINE_ORDINAL_ENTRIES,
+  DEFINE_ORDINAL_ENTRIES_ALTNAME,
+  DEFINE_ORDINAL_ENTRIES_APISET,
+  DEFINE_ORDINAL_MAP,
+  DEFINE_PROCNAME_ENTRIES,
+  DEFINE_PROCNAME_ENTRIES_ALTNAME,
+  DEFINE_PROCNAME_ENTRIES_APISET,
+  DEFINE_PROCNAME_MAP,
+  DLOENTRY,
+  DLOENTRY_APISET,
+  DLPENTRY,
+  DLPENTRY_APISET,
+
+  RpcEndExcept,
+
+  ActivatableClass,
+  ActivatableClassWithFactory,
+  ActivatableClassWithFactoryEx,
+  ActivatableStaticOnlyFactory,
+  ActivatableStaticOnlyFactoryEx,
+  CoCreatableClass,
+  CoCreatableClassWithFactory,
+  CoCreatableClassWithFactoryEx,
+]
+
+TypenameMacros: [
+  IFACEMETHOD,
+  STDMETHOD,
+  STDAPI_,
+]
+
+WhitespaceSensitiveMacros: [
+  __WI_CLANG_DISABLE_WARNING,

```

Comment:
Note that this has been custom added by me. Otherwise `clang-format` tries to separate `-` characters for Clang warning names